devimint/src/federation.rs

```diff
@@ -241,6 +249,22 @@ impl Federation {
         Ok(())
     }
 
+    pub async fn degrade_federation(&mut self, process_mgr: &ProcessManager) -> Result<()> {
+        let fed_size = process_mgr.globals.FM_FED_SIZE;
+        let offline_nodes = process_mgr.globals.FM_OFFLINE_NODES;
+        anyhow::ensure!(
+            fed_size > 3 * offline_nodes,
+            "too many offline nodes ({offline_nodes}) to reach consensus"
+        );
```

Comment:
Can the global `FM_OFFLINE_NODES` be changed after initialization? If not, we already check this in `dev_fed`?

Reply:
Similar in spirit to https://github.com/fedimint/fedimint/pull/4247#discussion_r1498500636

I think I was a bit overly cautious that a future caller might not have the safety check in `dev_fed` 😄